### PR TITLE
Don't record unspecified dependencies

### PR DESCRIPTION
**Pull request by mrvoorhe**
This gives us a way to punt on giving reasons to all of our marking behaviors.

### CODE REVIEW

File: src/linker/Linker/Tracer.cs [hunk 1]
```diff
@@ -70,6 +70,9 @@ bool IsRecordingEnabled ()
 
 		public void AddDirectDependency (object target, in DependencyInfo reason, bool marked)
 		{
+			if (reason.Kind == DependencyKind.Unspecified)
```
Comment (sbomer) [inline src/linker/Linker/Tracer.cs:73]: Could we instead let each dependency recorder decide what to do with `Unspecified` dependencies?

Comment (mrvoorhe) [reply src/linker/Linker/Tracer.cs]: Sure.  Done.

**Comment by mrvoorhe** [comment]
@marek-safar Does this look good to merge?

**Comment by mrvoorhe** [comment]
@sbomer Are you OK with the changes I made based on your comment?

**Review by vitek-karas — APPROVED** [review]

**Pull request merged**